lib/read_parsers.cc

```diff
@@ -58,8 +58,20 @@ void SeqAnParser::imprint_next_read(Read &the_read)
         ret = seqan::readRecord(the_read.name, the_read.sequence,
                                 the_read.quality, _private->stream);
         if (ret == 0) {
+            if (_num_reads == 0 && the_read.quality.length() != 0) {
+                _have_qualities = true;
+            }
             _num_reads++;
         }
+        if (the_read.sequence.length() == 0) {
```

Comment:
Cool, will do.

WRT `_num_reads`, wouldn't you then need to use `__sync_fetch_and_add`?